packages/edit-site/src/components/layout/router.js

```diff
@@ -24,23 +24,28 @@ export default function useLayoutAreas() {
 
 	// Regular page
 	if ( path === '/page' ) {
-		const isListLayout =
-			isCustom !== 'true' && ( ! layout || layout === 'list' );
 		return {
 			areas: {
-				content: window.__experimentalAdminViews ? (
-					<PagePages />
-				) : undefined,
-				preview: ( isListLayout ||
-					! window.__experimentalAdminViews ) && (
+				content: undefined,
+				preview: <Editor isLoading={ isSiteEditorLoading } />,
+			},
+			widths: {
+				content: undefined,
+			},
+		};
+	}
+
+	if ( path === '/pages' && window?.__experimentalAdminViews ) {
```

Comment:
Not sure if this is an issue of this PR or something external, but I see that the List layout is lost after refreshing, despite it being encoded in the URL. To test:

* Open your site at `/wp-admin/site-editor.php?path=%2Fpages&layout=list`
* Refresh
* Observe that you are redirected to `?path=%2Fpages` and there is a flicker as the layout changes from List to Table.

Reply:
This has been fixed at https://github.com/WordPress/gutenberg/pull/58081/commits/91427260f47ddd18a49bdf63627c9cb71bff1ace